containers.go

```diff
@@ -361,5 +369,28 @@ func runContainer(daemon *docker.Client, id string, logger log15.Logger, logfile
 		logger.Error("failed to start container", "error", err)
 		return nil, err
 	}
+	waiter = fdClosingWaiter{
+		CloseWaiter: waiter,
+		closers:     fdsToClose,
+		logger:      logger,
+	}
```

Comment:
Please refactor into 

```go
return fdClosingWaiter{
	CloseWaiter: waiter,
	closers:     fdsToClose,
	logger:      logger,
}, nil
```

I'd really like to avoid reassigning waiter with a completely different type that happens to implement the same interface.